pkg/config/config.go

```diff
@@ -20,48 +20,58 @@ var Cfg = new()
-	ServerAddress    string // Web server address
-	SlowLog          int    // Log operations slower than the specified time in ms. Default 1000ms
-	Version          string
+	DBBatchSize         int // Batch size used to write to DB. Default: 500
+	DBHealthCkeckPeriod int // Overrides pgxpool.Config{ HealthCheckPeriod } Default: 1 min
+	DBHost              string
+	DBMinConns          int // Overrides pgxpool.Config{ MinConns } Default: 0
+	DBMaxConns          int // Overrides pgxpool.Config{ MaxConns } Default: 20
+	DBMaxConnIdleTime   int // Overrides pgxpool.Config{ MaxConnIdleTime } Default: 30 min
+	DBMaxConnLifeTime   int // Overrides pgxpool.Config{ MaxConnLifetime } Default: 60 min
+	DBMaxConnLifeJitter int // Overrides pgxpool.Config{ MaxConnLifetimeJitter } Default: 2 min
+	DBName              string
+	DBPass              string
+	DBPort              int
+	DBUser              string
+	DevelopmentMode     bool
+	HTTPTimeout         int // Timeout for http server connections. Default: 5 min
+	KubeClient          *kubernetes.Clientset
+	KubeConfigPath      string
+	MaxBackoffMS        int // Maximum backoff in ms to wait after db connection error
+	PodName             string
+	PodNamespace        string
+	RediscoverRateMS    int    // time in MS we should check on cluster resource type
+	RequestLimit        int    // Max number of concurrent requests. Used to prevent from overloading the database
+	ServerAddress       string // Web server address
+	SlowLog             int    // Log operations slower than the specified time in ms. Default: 1 sec
+	Version             string
 }
 
 // Reads config from environment.
 func new() *Config {
 	conf := &Config{
-		DBBatchSize:     getEnvAsInt("DB_BATCH_SIZE", 500),
-		DBHost:          getEnv("DB_HOST", "localhost"),
-		DBMaxConns:      getEnvAsInt("DB_MAX_CONNS", int(20)), // Postgres has 100 conns. 20 allows scaling indexer
-		DBName:          getEnv("DB_NAME", ""),
-		DBPass:          getEnv("DB_PASS", ""),
-		DBPort:          getEnvAsInt("DB_PORT", 5432),
-		DBUser:          getEnv("DB_USER", ""),
-		DevelopmentMode: DEVELOPMENT_MODE,                    // Don't read ENV. See config_development.go to enable.
-		HTTPTimeout:     getEnvAsInt("HTTP_TIMEOUT", 300000), // 5 min
-		KubeConfigPath:  getKubeConfigPath(),
+		DBBatchSize: getEnvAsInt("DB_BATCH_SIZE", 500),
+		DBHost:      getEnv("DB_HOST", "localhost"),
+		// Postgres has 100 conns by default. Using 20 allows scaling indexer and api.
+		DBMaxConns:          getEnvAsInt("DB_MAX_CONNS", 20),                   // 20 - Overrides pgxpool default
+		DBMaxConnLifeJitter: getEnvAsInt("DB_MAX_CONN_LIFE_JITTER", 2*60*1000), // 2 min - Overrides pgxpool default
+		DBMaxConnIdleTime:   getEnvAsInt("DB_MAX_CONN_IDLE_TIME", 30*60*1000),  // 30 min - Default for pgxpool.Config
+		DBMaxConnLifeTime:   getEnvAsInt("DB_MAX_CONN_IDLE_TIME", 60*60*1000),  // 60 min - Default for pgxpool.Config
```

Comment:
```suggestion
		DBMaxConnLifeTime:   getEnvAsInt("DB_MAX_CONN_LIFE_TIME", 60*60*1000),  // 60 min - Default for pgxpool.Config
```

Reply:
oops copy/paste error. Pushed the change.